packages/workbox-precaching/src/utils/cacheWrapper.ts

```diff
@@ -0,0 +1,111 @@
+/*
+  Copyright 2018 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {WorkboxPlugin} from 'workbox-core/types.js';
+import {Strategy} from 'workbox-strategies/Strategy.js';
+import {StrategyHandler} from 'workbox-strategies/StrategyHandler.js';
+import '../_version.js';
+
+
+interface MatchWrapperOptions {
+  cacheName: string;
+  request: Request;
+  event?: ExtendableEvent;
+  plugins?: WorkboxPlugin[];
+  matchOptions?: CacheQueryOptions;
+}
+
+interface PutWrapperOptions extends MatchWrapperOptions {
+  response: Response;
+}
+
+class CacheStrategy extends Strategy {
+  async _handle(request: Request, handler: StrategyHandler) {
+    // Will not be called, but needed to make TypeScript happy...
+    return handler.fetch(request);
```

Comment:
Do you want to `throw` here or something to actually ensure that it's not called?

If for some reason it _is_ called, `return handler.fetch(request)` seems like it would be pretty hard to debug just from looking at DevTools.

Reply:
That's what I originally had, but then TypeScript was not happy. Since I plan to remove this code once I update the `workbox-precaching` package to use a custom strategy, I think it's OK to leave this as is for now.